lib/Monads/trace/Trace_Monad.thy

```diff
@@ -923,81 +784,66 @@ abbreviation (input)
   "validIEsat f P R G Q E \<equiv> validIE f P R G Q E"
  *)
 text \<open>
-  The following two instantiations are convenient to separate reasoning
-  for exceptional and normal case.
-\<close>
-definition
-  validE_R :: "('s \<Rightarrow> bool) \<Rightarrow> ('s, 'e + 'a) tmonad \<Rightarrow>
-               ('a \<Rightarrow> 's \<Rightarrow> bool) \<Rightarrow> bool"
-   ("\<lbrace>_\<rbrace>/ _ /\<lbrace>_\<rbrace>, -")
-where
+  The following two instantiations are convenient to separate reasoning for exceptional and
+  normal case.\<close>
+(* Narrator: they are in fact not convenient, and are now considered a mistake that should have
+             been an abbreviation instead. *)
```

Comment:
does this indicate this should be a FIXME?

Reply:
Maybe? This comment is copied verbatim from the nondet monad though, so I'd prefer not to change it as part of this work.

Reply:
FIXME would indicate intent to change it. I think we probably don't want to change this, but we should also record that it probably wasn't the optimal decision to go with this in the first place.

Reply:
To clarify, there actually is a FIXME on the definition in the following line, saying `(* FIXME lib: this should be an abbreviation *)`
See https://github.com/seL4/l4v/blob/master/lib/Monads/nondet/Nondet_VCG.thy#L66